src/gui/qgsmapcanvas.cpp

```diff
@@ -950,6 +951,18 @@ void QgsMapCanvas::rendererJobFinished()
     mRenderedItemResults = std::move( renderedItemResults );
     mRenderedItemResultsOutdated = false;
 
+    for ( const QgsRenderedItemDetails *details : mRenderedItemResults->renderedItems() )
+    {
+      // if the min-max of a raster have changed - update the style and the legend
+      const QgsRenderedCalculatedResults *renderedDetails = dynamic_cast<const QgsRenderedCalculatedResults *>( details );
+      QgsMapLayer *layer = renderedDetails ? QgsProject::instance()->mapLayer( renderedDetails->layerId() ) : nullptr;
+      if ( layer )
+      {
+        emit calculatedRenderDetailsChanged();
+        break;
+      }
+    }
```

Comment:
I'd lean toward moving this logic to qgisapp.cpp instead, and using the existing renderComplete signal to trigger the check for changed results. That'd avoid the need for adding the new calculatedRenderDetailsChanged signal to the (already large) QgsMapCanvas API.

Reply:
Done.